angular/whereami/src/app/views/create-game/hand-picked/hand-picked-manager.component.ts

```diff
@@ -69,12 +70,51 @@ export class HandPickedManagerComponent {
         this.parent.googleMap.addListener("click", (event: google.maps.MapMouseEvent | google.maps.IconMouseEvent) => {
           this.changed = true;
           const radius = 50000 / Math.pow(1.6, <number>this.parent.getZoom());
-          this.streetViewService.getPanorama({location: event.latLng, radius: radius}, this.processPano);
+          this.addMarkerFromPanorama(event.latLng!, radius);
         });
       }
     }
   }
 
+  public addMarker(loc: google.maps.LatLng, name: string) {
+    if (!this.locationExists(loc)) {
+      const marker = new this.google_ns.maps.Marker({
+        position: loc,
+        map: this.parent.googleMap,
+        title: name,
+        draggable: true,
+      });
+      this._markers.push(marker);
+      this.activateMarker(marker);
+      marker.addListener("click", () => this.activateMarker(marker));
+      marker.addListener("dragstart", () => this.onDragStart(marker))
+      marker.addListener("dragend", () => this.onDragEnd(marker));
+    }
+  }
+
+  private locationExists(latLng: google.maps.LatLng) {
+    for (let markerId in this._markers) {
+        var presentMarker = this._markers[markerId];
+        if (this.activeMarker === presentMarker) {
+            continue;
+        }
+        var distance = google.maps.geometry.spherical.computeDistanceBetween(presentMarker.getPosition()!, latLng);
```

Comment:
please use the google_ns injection token. 
I tried to at least somewhat decuple components for e.g. unit testing. 

PS: For type hints it does not matter whether you use google_ns or google.